cluster-api/CONTRIBUTING.md

```diff
@@ -0,0 +1,60 @@
+# Cluster API Development Guide
+
+## Cloud Provider Dev Guide
+
+### Overview
+
+The Cluster API is a Kubernetes project to bring declarative, Kubernetes-style APIs to cluster creation, configuration, and management. It provides optional, additive functionality on top of core Kubernetes.
+
+This document is meant to help OSS contributors implement support for cloud providers.
+
+As part of adding support for a provider (cloud or on-prem), you will need to:
+
+1.  Create tooling that conforms to the Cluster API (described further below)
+1.  A machine controller that can run independent of the cluster. This controller should handle the lifecycle of the machines, whether it's run inside the master in a pod, in an external API server, or even on your local workstation.
```

Comment:
Should this be 2. ?

Reply:
machine controller has nothing with API server. The whole sentence should be simplified as in-cluster or out-cluster.

Reply:
@jessicaochen - (Github's) markdown renders the list just fine when it's numbered.

@medinatiger Done.